### ISSUE TITLE
trying to diagnose login issues

### ISSUE THREAD

**Issue by fossfreedom** [solus-project/budgie-desktop#297]
What's the best way to try and diagnose why I get a blank screen with just the wallpaper - after 30 seconds I then get dumped back to the login screen.

I've looked at /var/log/syslog - the last 200 lines has various budgie type warnings - any of these makes sense?

Should I remake the .deb with DEBUG on and relook at syslog?  i.e. ./configure --enable-debug ?

**Comment by ikeydoherty** [comment]
Make sure you have gnome-settings-daemon and gnome-session

**Comment by fossfreedom** [comment]
gnome-session wasn't installed but gnome-settings-daemon was.  So, installed gnome-session, Rebooted - no change - just the wallpaper.

**Comment by ikeydoherty** [comment]
I'd check .xsession-errors or `journalctl -ae` or whichever the equivalent is on Debian. Clearly something is missing

**Comment by fossfreedom** [comment]
.xsession-errors - nothing really here

<pre>
dad@dad-VirtualBox:~$ tail -200 .xsession-errors
openConnection: connect: No such file or directory
cannot connect to brltty at :0
Service 'org.kde.kaccessibleapp' does not exist.
</pre>

journalctl -ae --full

<pre>
Jan 02 23:25:09 dad-VirtualBox org.a11y.atspi.Registry[3731]: SpiRegistry daemon is running with well-known name - org.a11y.atspi.Registry
Jan 02 23:25:09 dad-VirtualBox gnome-session[3715]: gnome-session-is-accelerated: llvmpipe detected.
Jan 02 23:25:09 dad-VirtualBox org.gnome.ScreenSaver[3627]: ** (gnome-screensaver:3765): WARNING **: Couldn't get presence status: The name org.gnome.SessionManager was not provided by any .service files
Jan 02 23:25:09 dad-VirtualBox org.ayatana.bamf[3627]: initctl: Job is already running: bamfdaemon
Jan 02 23:25:09 dad-VirtualBox gnome-session[3715]: gnome-session-binary[3715]: WARNING: Could not parse desktop file budgie-desktop-polkit-gnome-authentication-agent-1.desktop or it references a not found TryExec binary
Jan 02 23:25:09 dad-VirtualBox gnome-session-binary[3715]: WARNING: Could not parse desktop file budgie-desktop-polkit-gnome-authentication-agent-1.desktop or it references a not found TryExec binary
Jan 02 23:25:09 dad-VirtualBox gnome-session[3715]: gnome-session-binary[3715]: WARNING: Could not parse desktop file telepathy-indicator.desktop or it references a not found TryExec binary
Jan 02 23:25:09 dad-VirtualBox gnome-session-binary[3715]: WARNING: Could not parse desktop file telepathy-indicator.desktop or it references a not found TryExec binary
Jan 02 23:25:10 dad-VirtualBox gnome-keyring-daemon[3459]: The PKCS#11 component was already initialized
Jan 02 23:25:10 dad-VirtualBox gnome-session[3715]: SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
Jan 02 23:25:10 dad-VirtualBox dbus[512]: [system] Activating via systemd: service name='org.bluez' unit='dbus-org.bluez.service'
Jan 02 23:25:10 dad-VirtualBox gnome-keyring-daemon[3459]: The Secret Service was already initialized
Jan 02 23:25:10 dad-VirtualBox gnome-session[3715]: SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
Jan 02 23:25:10 dad-VirtualBox gnome-keyring-daemon[3459]: The SSH agent was already initialized
Jan 02 23:25:10 dad-VirtualBox gnome-session[3715]: SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
</pre>

**Comment by fossfreedom** [comment]
ok - have got past the polkit issue - the path in /etc/xdg/autostart/budgie-desktop-polkit-gnome-authentication-agent-1.desktop is wrong (or at least I need to correct the shipped .desktop file if there is one in the deb package) - manually corrected this and this error is no longer shown.

I don't think the telepathy-indicator is relevant - again - install telepathy-indicator and this message has gone.

this just leaves this in journalctl -ae --full

<pre>
Jan 03 00:13:30 dad-VirtualBox gnome-session[7521]: pci id for fd 11: 80ee:beef, driver (null)
Jan 03 00:13:30 dad-VirtualBox gnome-session[7521]: libGL error: core dri or dri2 extension not found
Jan 03 00:13:30 dad-VirtualBox gnome-session[7521]: libGL error: failed to load driver: vboxvideo
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-wm:7742): GLib-GObject-WARNING **: invalid (NULL) pointer instance
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-wm:7742): GLib-GObject-CRITICAL **: g_signal_connect_object: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-panel:7750): GLib-GObject-WARNING **: cannot register existing type 'BudgieDesktopManager'
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-panel:7750): GLib-CRITICAL **: g_once_init_leave: assertion 'result != 0' failed
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-panel:7750): GLib-GObject-CRITICAL **: g_param_spec_object: assertion 'g_type_is_a (object_type, G_TYPE_OBJECT)' failed
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-panel:7750): GLib-GObject-CRITICAL **: g_object_class_install_property: assertion 'G_IS_PARAM_SPEC (pspec)' failed
Jan 03 00:13:31 dad-VirtualBox gnome-session[7521]: (budgie-panel:7750): GLib-GObject-CRITICAL **: g_object_new_valist: object class 'BudgieRaven' has no property named 'manager'
</pre>

**Comment by ikeydoherty** [comment]
The polkit file is going to disappear soon. I'll have to see why Vala/GLib is being a dipshit about the BudgieDesktopManager API.

**Comment by ikeydoherty** [comment]
polkit file gone

**Comment by fossfreedom** [comment]
looking back at the compilation of budgie via the deb there are quite a few warnings about incompatible pointer type assignments and other stuff - see line 4064/3875 as a couple of examples.

Anything here that is a worry that maybe causing budgie-wm/panel to throw those critical errors above?

- https://dl.dropboxusercontent.com/u/3575524/budgie2.txt

**Comment by ikeydoherty** [comment]
No it's just a combination of two factors. The horror-show that is the internals of gobject (aka struct casting) and the abomination that is Vala's autogenerated C code.

**Comment by fossfreedom** [comment]
@ikeydoherty - do you have handy a list of files and their file locations where budgie-desktop is installed?

What I want to check is to see if my debian package is building correctly and putting stuff in the correct place when the .deb is actually installed.

**Comment by ikeydoherty** [comment]
File list for budgie-desktop:
```
/etc/xdg/autostart/budgie-desktop-nm-applet.desktop
/usr/bin/budgie-desktop
/usr/bin/budgie-panel
/usr/bin/budgie-polkit-dialog
/usr/bin/budgie-run-dialog
/usr/bin/budgie-wm
/usr/lib64/budgie-desktop/Gvc-1.0.typelib
/usr/lib64/budgie-desktop/plugins/BudgieMenu.plugin
/usr/lib64/budgie-desktop/plugins/ClockApplet.plugin
/usr/lib64/budgie-desktop/plugins/IconTasklistApplet.plugin
/usr/lib64/budgie-desktop/plugins/LockKeysApplet.plugin
/usr/lib64/budgie-desktop/plugins/NotificationsApplet.plugin
/usr/lib64/budgie-desktop/plugins/SeparatorApplet.plugin
/usr/lib64/budgie-desktop/plugins/ShowDesktopApplet.plugin
/usr/lib64/budgie-desktop/plugins/SpacerApplet.plugin
/usr/lib64/budgie-desktop/plugins/StatusApplet.plugin
/usr/lib64/budgie-desktop/plugins/TasklistApplet.plugin
/usr/lib64/budgie-desktop/plugins/TrayApplet.plugin
/usr/lib64/budgie-desktop/plugins/WorkspacesApplet.plugin
/usr/lib64/budgie-desktop/plugins/libbudgiemenuapplet.so
/usr/lib64/budgie-desktop/plugins/libclockapplet.so
/usr/lib64/budgie-desktop/plugins/libicontasklistapplet.so
/usr/lib64/budgie-desktop/plugins/liblockkeysapplet.so
/usr/lib64/budgie-desktop/plugins/libnotificationsapplet.so
/usr/lib64/budgie-desktop/plugins/libseparatorapplet.so
/usr/lib64/budgie-desktop/plugins/libshowdesktopapplet.so
/usr/lib64/budgie-desktop/plugins/libspacerapplet.so
/usr/lib64/budgie-desktop/plugins/libstatusapplet.so
/usr/lib64/budgie-desktop/plugins/libtasklistapplet.so
/usr/lib64/budgie-desktop/plugins/libtrayapplet.so
/usr/lib64/budgie-desktop/plugins/libworkspacesapplet.so
/usr/lib64/girepository-1.0/Budgie-1.0.typelib
/usr/lib64/libbudgie-plugin.so.0
/usr/lib64/libbudgie-plugin.so.0.0.0
/usr/lib64/libbudgietheme.so.0
/usr/lib64/libbudgietheme.so.0.0.0
/usr/lib64/libraven.so.0
/usr/lib64/libraven.so.0.0.0
/usr/share/applications/budgie-panel.desktop
/usr/share/applications/budgie-polkit.desktop
/usr/share/applications/budgie-wm.desktop
/usr/share/gir-1.0/Budgie-1.0.gir
/usr/share/glib-2.0/schemas/com.solus-project.budgie-menu.gschema.xml
/usr/share/glib-2.0/schemas/com.solus-project.budgie-panel.gschema.xml
/usr/share/glib-2.0/schemas/com.solus-project.budgie.wm.gschema.xml
/usr/share/glib-2.0/schemas/com.solus-project.icon-tasklist.gschema.xml
/usr/share/glib-2.0/schemas/com.solus-project.spacer.gschema.xml
/usr/share/gnome-session/sessions/budgie-desktop.session
/usr/share/icons/hicolor/scalable/actions/notification-alert-symbolic.svg
/usr/share/locale/ca/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/cs/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/de/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/de_DE/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/en_GB/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/eo/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/es/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/es_AR/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/fi/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/fr/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/fr_FR/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/id/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/id_ID/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/it/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/it_IT/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/ja_JP/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/lt/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/nl_BE/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/pl/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/pt/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/pt_BR/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/ro/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/ru/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/sr_RS@latin/LC_MESSAGES/budgie-desktop.mo
/usr/share/locale/tr/LC_MESSAGES/budgie-desktop.mo
/usr/share/xsessions/budgie-desktop.desktop
```

File list for budgie-desktop-devel:
```
/usr/include/BudgiePlugin.h
/usr/include/BudgieRaven.h
/usr/include/BudgieSession.h
/usr/include/BudgieToplevel.h
/usr/lib64/libbudgie-plugin.so
/usr/lib64/libbudgietheme.so
/usr/lib64/libraven.so
/usr/lib64/pkgconfig/budgie-1.0.pc
/usr/share/vala/vapi/budgie-1.0.deps
/usr/share/vala/vapi/budgie-1.0.vapi
```

These file lists are from the Solus packaging

**Comment by ikeydoherty** [comment]
(Also that's 10.2.2 not git)

**Comment by fossfreedom** [comment]
thanks - looks like I'm packaging the same files in the same file locations.

Any other way I can get more debug info out of the various components on startup?

**Comment by ikeydoherty** [comment]
Edit the /usr/bin/budgie-desktop file to pass --debug to gnome-session

**Comment by fossfreedom** [comment]
thanks - any of this make any sense?

https://dl.dropboxusercontent.com/u/3575524/budgie3.txt

**Comment by ikeydoherty** [comment]
```
Jan 06 00:07:05 dad-VirtualBox gnome-session-binary[5604]: WARNING: Application 'budgie-panel.desktop' failed to register before timeout
Jan 06 00:07:05 dad-VirtualBox gnome-session-binary[5604]: Unrecoverable failure in required component budgie-panel.desktop
Jan 06 00:07:05 dad-VirtualBox gnome-session-binary[5604]: CRITICAL: We failed, but the fail whale is dead. Sorry....```

According to this, you definitely have issues :) 

I can't even find that string in gnome-session so I've no idea where it comes from. The fact the fail-whale won't even start is a definite sign that something is wrong.

At this point I'd even recommend trying to install GNOME Shell to see if it brings in any missing core dependencies, as I know Debian has a tendency to split packages up into lots of tiny pieces and make my life hard

**Comment by fossfreedom** [comment]
hmm - thanks.  Tried as you have said - now have gnome-shell, unity and budgie installed.

Same error as you have found above - whatever that whale is ... its dead.

**Comment by ikeydoherty** [comment]
o_O Alright can you see if you're able to launch budgie-panel from one of the other desktops? Just to ensure it's not broken :D

**Comment by fossfreedom** [comment]
From Unity

<pre>
dad@dad-VirtualBox:~$ budgie-panel &
[1] 10221
dad@dad-VirtualBox:~$ 
(budgie-panel:10221): GLib-GObject-WARNING **: cannot register existing type 'BudgieDesktopManager'

(budgie-panel:10221): GLib-CRITICAL **: g_once_init_leave: assertion 'result != 0' failed

(budgie-panel:10221): GLib-GObject-CRITICAL **: g_param_spec_object: assertion 'g_type_is_a (object_type, G_TYPE_OBJECT)' failed

(budgie-panel:10221): GLib-GObject-CRITICAL **: g_object_class_install_property: assertion 'G_IS_PARAM_SPEC (pspec)' failed

(budgie-panel:10221): GLib-GObject-CRITICAL **: g_object_new_valist: object class 'BudgieRaven' has no property named 'manager'
</pre>

If I launch budgie-desktop I get to the wallpaper only screen and I can use CTRL+ALT+T to launch a terminal.

<pre>
dad@dad-VirtualBox:~$ budgie-panel &
[1] 10650
dad@dad-VirtualBox:~$ ** Message: manager.vala:898: Another panel is already running. Use --replace to replace it

[1]+  Done                    budgie-panel
dad@dad-VirtualBox:~$ budgie-panel --replace &
[1] 10654
dad@dad-VirtualBox:~$ 
(budgie-panel:10654): GLib-GObject-WARNING **: cannot register existing type 'BudgieDesktopManager'

(budgie-panel:10654): GLib-CRITICAL **: g_once_init_leave: assertion 'result != 0' failed

(budgie-panel:10654): GLib-GObject-CRITICAL **: g_param_spec_object: assertion 'g_type_is_a (object_type, G_TYPE_OBJECT)' failed

(budgie-panel:10654): GLib-GObject-CRITICAL **: g_object_class_install_property: assertion 'G_IS_PARAM_SPEC (pspec)' failed

(budgie-panel:10654): GLib-GObject-CRITICAL **: g_object_new_valist: object class 'BudgieRaven' has no property named 'manager'

</pre>

Note - I can also right click the wallpaper and see the menu options "change background" and "Settings" - both invoke what they should so - is that gnome-session or some part of budgie window manager running?

The terminal has a background shadow - so it looks like the window manager/compositor of somesort is running.

**Comment by ikeydoherty** [comment]
The right click stuff is budgie-wm.

Looking at the error there's a deep issue in the type system from your vala/glib combination. I'll have to do some workarounds to get it up and running.

**Comment by fossfreedom** [comment]
@ikeydoherty 

  I am right on this - a GPL project like sparky are supposed to supply their source when requested?

http://sparkylinux.org/forum/index.php/topic,3397.0.html

**Comment by ikeydoherty** [comment]
I've had a look at the LiveCD, Budgie is unmodified entirely in Sparky Linux.

The customisations are (unfortunately) made by use of `/etc/skel/.config/dconf` tree, which populates the new user account.

I'd actually like to talk with the developer on improving this, they should be using gschema overrides to achieve their default L&F

As for how they distribute their package, well, I'm not a lawyer, so unless I start getting screwed over, I'd like to stay out of it.

**Comment by fossfreedom** [comment]
ok - I had a feeling the project was doing that.  Will let you know if the package they use becomes available.

**Comment by fossfreedom** [comment]
ok - think I understand now what sparky have done.

They compile and install manually via ./autogen --prefix=/usr

Then they create a .deb from the prebuilt binaries and ship that.  

Its not a method Debian nor Ubuntu etc allows - you must build in a relative folder way (i.e. debuild) and the built package moves the built binaries stored in the debian/tmp/usr/... folders to their correct location 

i.e. ./debian/tmp/usr/share/budgie-desktop/xyz not /usr/share/budgie-desktop/xyz

I guess there is something in budgie-desktop compilation that doesnt like this relative folder building method e.g. /usr/share/gir-1.0/Budgie-1.0.gir is 35.7Kb when manually built but via a debian build - Budgie-1.0.gir is only 6kb in size.

Any thoughts @ikeydoherty ?

**Comment by fossfreedom** [comment]
ok - going to close this.  I cannot fix this under Ubuntu 16.04.

Whatever ubuntu is doing whilst building a .deb package is not happening if I build under debian testing - debian testing works fine.  I can even take the built debian testing package and install that under Ubuntu 16.04!  everything works ok.

It's a shame - I wanted to give budgie-desktop the massive attention it deserves by the huge ubuntu + ubuntu derivative users :(

Anyway good luck with future versions of budgie @ikeydoherty
